hnn_core/network_models.py

```diff
@@ -0,0 +1,263 @@
+
+    # layer2 Basket -> layer2 Pyr
+    src_cell = 'L2_basket'
+    target_cell = 'L2_pyramidal'
+    lamtha = 50.
+    loc = 'soma'
+    for receptor in ['gabaa', 'gabab']:
+        key = f'gbar_L2Basket_L2Pyr_{receptor}'
+        weight = net._params[key]
+        net.add_connection(
+            src_cell, target_cell, loc, receptor, weight, delay, lamtha)
+
+    # layer5 Basket -> layer5 Pyr
+    src_cell = 'L5_basket'
+    target_cell = 'L5_pyramidal'
+    lamtha = 70.
+    loc = 'soma'
+    for receptor in ['gabaa', 'gabab']:
+        key = f'gbar_L5Basket_{_short_name(target_cell)}_{receptor}'
+        weight = net._params[key]
+        net.add_connection(
+            src_cell, target_cell, loc, receptor, weight, delay, lamtha)
+
+    # layer2 Pyr -> layer5 Pyr
+    src_cell = 'L2_pyramidal'
+    lamtha = 3.
+    receptor = 'ampa'
+    for loc in ['proximal', 'distal']:
+        key = f'gbar_L2Pyr_{_short_name(target_cell)}'
+        weight = net._params[key]
+        net.add_connection(
+            src_cell, target_cell, loc, receptor, weight, delay, lamtha)
+
+    # layer2 Basket -> layer5 Pyr
+    src_cell = 'L2_basket'
+    lamtha = 50.
+    key = f'gbar_L2Basket_{_short_name(target_cell)}'
+    weight = net._params[key]
+    loc = 'distal'
+    receptor = 'gabaa'
+    net.add_connection(
+        src_cell, target_cell, loc, receptor, weight, delay, lamtha)
+
+    # xx -> layer2 Basket
+    src_cell = 'L2_pyramidal'
+    target_cell = 'L2_basket'
+    lamtha = 3.
+    key = f'gbar_L2Pyr_{_short_name(target_cell)}'
+    weight = net._params[key]
+    loc = 'soma'
+    receptor = 'ampa'
+    net.add_connection(
+        src_cell, target_cell, loc, receptor, weight, delay, lamtha)
+
+    src_cell = 'L2_basket'
+    lamtha = 20.
+    key = f'gbar_L2Basket_{_short_name(target_cell)}'
+    weight = net._params[key]
+    loc = 'soma'
+    receptor = 'gabaa'
+    net.add_connection(
+        src_cell, target_cell, loc, receptor, weight, delay, lamtha)
+
+    # xx -> layer5 Basket
+    src_cell = 'L5_basket'
+    target_cell = 'L5_basket'
+    lamtha = 20.
+    loc = 'soma'
+    receptor = 'gabaa'
+    key = f'gbar_L5Basket_{_short_name(target_cell)}'
+    weight = net._params[key]
+    net.add_connection(
+        src_cell, target_cell, loc, receptor, weight, delay, lamtha,
+        allow_autapses=False)
+
+    src_cell = 'L5_pyramidal'
+    lamtha = 3.
+    key = f'gbar_L5Pyr_{_short_name(target_cell)}'
+    weight = net._params[key]
+    loc = 'soma'
+    receptor = 'ampa'
+    net.add_connection(
+        src_cell, target_cell, loc, receptor, weight, delay, lamtha)
+
+    src_cell = 'L2_pyramidal'
+    key = f'gbar_L2Pyr_{_short_name(target_cell)}'
+    weight = net._params[key]
+    loc = 'soma'
+    receptor = 'ampa'
+    net.add_connection(
+        src_cell, target_cell, loc, receptor, weight, delay, lamtha)
+
+    return net
+
+
+def law_model(params=None, add_drives_from_params=False):
```

Comment:
is this function covered by any tests?

Reply:
Also: `law_2021_model` to conform with convention

Reply:
Added some tests but I'm not sure how extensive they should be. For example, is it necessary to have a separate `dpl.txt` for each model we create? Or just a few checks like I have added to make sure things are breaking.